### PR TITLE
Add Trino 414 release notes

### LINKED ISSUE #17002: Release notes for 414

### PR DESCRIPTION
**Pull request by colebow**
## Description

Assemble the release notes for the upcoming Trino 414 release. Fixes #17002 

## Release notes

(x) This is not user-visible or docs only and no release notes are required.

## Verification for each pull request

Format: maintainer, PR/issue number, ✅ / ❌ rn ✅ / ❌ docs
✅ rn - release note added and verified, or assessed to be not necessary, set to ❌ rn before completion
✅ docs - need for docs assessed and merged, or assessed to be not necessary, set to ❌ docs before completion

## 13 Apr 2023

* #16992 ✅ rn ✅ docs
* #16738 ✅ rn ✅ docs
* #17018 ✅ rn ✅ docs
* #16995 ✅ rn ✅ docs
* #16991 ✅ rn ✅ docs
* #16833 ✅ rn ✅ docs
* #16854 ✅ rn ✅ docs
* #16998 ✅ rn ✅ docs
* #16850 ✅ rn ✅ docs
* #17010 ✅ rn ✅ docs
* #16962 ✅ rn ✅ docs
* #17025 ✅ rn ✅ docs
* #17028 ✅ rn ✅ docs

## 14 Apr 2023

* #16885 ✅ rn ✅ docs
* #17022 ✅ rn ✅ docs
* #16988 ✅ rn ✅ docs
* #17029 ✅ rn ✅ docs
* #16877 ✅ rn ✅ docs
* #16716 ✅ rn ✅ docs
* #16993 ✅ rn ✅ docs

## 15 Apr 2023

* #16036 ✅ rn ✅ docs
* #17036 ✅ rn ✅ docs
* #17030 ✅ rn ✅ docs
* #17037 ✅ rn ✅ docs
* #17039 ✅ rn ✅ docs
* #16815 ✅ rn ✅ docs
* #17041 ✅ rn ✅ docs

## 16 Apr 2023

* #16905 ✅ rn ✅ docs
* #17040 ✅ rn ✅ docs
* #17031 ✅ rn ✅ docs

## 17 Apr 2023

* #17047 ✅ rn ✅ docs
* #17048 ✅ rn ✅ docs
* #17000 ✅ rn ✅ docs
* #17007 ✅ rn ✅ docs
* #16704 ✅ rn ✅ docs
* #17044 ✅ rn ✅ docs
* #16950 ✅ rn ❌ docs
* #16183 ✅ rn ✅ docs

## 18 Apr 2023

* #17034 ✅ rn ✅ docs
* #17074 ✅ rn ✅ docs
* #17083 ✅ rn ✅ docs
* #14937 ✅ rn ✅ docs
* #17099 ✅ rn ✅ docs
* #17098 ✅ rn ✅ docs
* #16843 ✅ rn ✅ docs
* #16557 ✅ rn ✅ docs
* #17084 ✅ rn ✅ docs
* #17086 ✅ rn ✅ docs
* #16867 ✅ rn ✅ docs
* #17066 ✅ rn ✅ docs
* #17111 ✅ rn ✅ docs
* #17097 ✅ rn ✅ docs

## 19 Apr 2023

* #17013 ✅ rn ✅ docs
* #16860 ✅ rn ❌ docs
* #17105 ✅ rn ✅ docs
* #12653 ✅ rn ✅ docs

### CODE REVIEW

File: docs/src/main/sphinx/release/release-414.md [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+* Add support for Hive external tables in the `migrate` table procedure. ({issue}`16704`)
+
+## Kafka connector
+
+* Fix query failure when a Kafka topic contains tombstones (messages with a
+  ``NULL`` value). ({issue}`16962`)
+
+## Kudu connector
+
+* Fix query failure when merging two tables that were created by
```
Comment (martint) [inline docs/src/main/sphinx/release/release-414.md:39]: `CREATE TABLE ... AS SELECT ...`

File: docs/src/main/sphinx/release/release-414.md [hunk 2]
```diff
@@ -0,0 +1,52 @@
+# Release 414 (19 Apr 2023)
+
+## General
+
+* Add [recursive member access](json_descendant_member_accessor) to the
+  [JSON path language](json-path-language). ({issue}`16854`)
+* Add the [`sequence()`](built_in_table_functions) table function. ({issue}`16716`)
+* Add support for progress estimates when
+  [fault-tolerant execution](/admin/fault-tolerant-execution) is enabled. ({issue}`13072`)
+* Add support for `CUBE` and `ROLLUP` with composite sets ({issue}`16981`)
```
Comment (mosabua) [inline docs/src/main/sphinx/release/release-414.md]:
```suggestion
* Add support for `CUBE` and `ROLLUP` with composite sets. ({issue}`16981`)
```

File: docs/src/main/sphinx/release/release-414.md [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+* Add experimental support for tracing using [OpenTelemetry](https://opentelemetry.io/).
+  This can be enabled by setting the `tracing.enabled` configuration property to
+  `true` and configuring the
+  [OLTP/gRPC endpoint](https://opentelemetry.io/docs/reference/specification/protocol/otlp/)
+  by setting the `tracing.exporter.endpoint` configuration property if needed. ({issue}`16950`)
```
Comment (mosabua) [inline docs/src/main/sphinx/release/release-414.md:15]:
```suggestion
  by setting the `tracing.exporter.endpoint` configuration property. ({issue}`16950`)
```

File: docs/src/main/sphinx/release/release-414.md [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+
+* Add [recursive member access](json_descendant_member_accessor) to the
+  [JSON path language](json-path-language). ({issue}`16854`)
+* Add the [`sequence()`](built_in_table_functions) table function. ({issue}`16716`)
+* Add support for progress estimates when
+  [fault-tolerant execution](/admin/fault-tolerant-execution) is enabled. ({issue}`13072`)
+* Add support for `CUBE` and `ROLLUP` with composite sets ({issue}`16981`)
+* Add experimental support for tracing using [OpenTelemetry](https://opentelemetry.io/).
+  This can be enabled by setting the `tracing.enabled` configuration property to
+  `true` and configuring the
```
Comment (mosabua) [inline docs/src/main/sphinx/release/release-414.md:13]:
```suggestion
  `true`, and optionally configuring the
```

File: docs/src/main/sphinx/release/release-414.md [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+  [OLTP/gRPC endpoint](https://opentelemetry.io/docs/reference/specification/protocol/otlp/)
+  by setting the `tracing.exporter.endpoint` configuration property if needed. ({issue}`16950`)
+* Fix query failure for recursive queries involving lambda expressions. ({issue}`16989`)
+* Fix incorrect results when using the {func}`sequence` function with values
```
Comment (mosabua) [inline docs/src/main/sphinx/release/release-414.md:18]: did you need to change this to the ref?

**Review by mosabua — CHANGES_REQUESTED** [review]

File: docs/src/main/sphinx/release/release-414.md (hunk 5 above)
Comment (colebow) [reply docs/src/main/sphinx/release/release-414.md:18]: No, this one's correct. The sequence table function mentioned above is the anchor, but this is a fix for the existing function.

File: docs/src/main/sphinx/release/release-414.md [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+* Disallow reading tables with [deletion vectors](https://github.com/delta-io/delta/blob/master/PROTOCOL.md#deletion-vectors).
+  Previously, this returned incorrect results. ({issue}`16884`)
+
+## Iceberg connector
+
+* Add support for Hive external tables in the `migrate` table procedure. ({issue}`16704`)
+
+## Kafka connector
+
+* Fix query failure when a Kafka topic contains tombstones (messages with a
```
[the remark this answers is not in the document]
Comment (colebow) [reply docs/src/main/sphinx/release/release-414.md:35]: They're generally referred to as tombstones, so I feel like it's helpful for someone who's familiar with the term and might not understand what we're getting at if we don't mention them.

File: docs/src/main/sphinx/release/release-414.md [hunk 7]
```diff
@@ -0,0 +1,53 @@
...
+  [OLTP/gRPC endpoint](https://opentelemetry.io/docs/reference/specification/protocol/otlp/)
+  by setting the `tracing.exporter.endpoint` configuration property. ({issue}`16950`)
+* Improve performance for certain queries that produce no values. ({issue}`15555`, {issue}`16515`)
+* Fix query failure for recursive queries involving lambda expressions. ({issue}`16989`)
+* Fix incorrect results when using the {func}`sequence` function with values
+  greater than 2<sup>31</sup> (about 2.1 billion). ({issue}`16742`)
+
+## Delta Lake connector
+
+* Add support for `SELECT`, `INSERT`, `UPDATE`, and `DELETE` operations on
```
Comment (mosabua) [inline docs/src/main/sphinx/release/release-414.md:23]: I assume this also works for MERGE .. but dont think we need to add that.

**Review by mosabua — APPROVED** [review]
lgtm now

File: docs/src/main/sphinx/release/release-414.md (hunk 7 above)
Comment (ebyhr) [reply docs/src/main/sphinx/release/release-414.md:30]: Please remove `SELECT`. Yes, the change contains `MERGE` support.

**Comment by martint** [comment]
Applied changes and merged as a2a05043ac9167e7486d35c9523cd9b5c1b6ccff

**Pull request closed without merging**